app/classes/importers/photosynthesis_measurement_importer.rb

```diff
@@ -0,0 +1,65 @@
+class PhotosynthesisMeasurementImporter < RowImporter
+
+  def object
+    @pm
+  end
+
+  def read_row(values, logger)
+    @pm = find_or_new({
+      #########################################################################
+      #
+      # The code reader is going to need to do a      CSP -> tree code lookup
+      # All the data we will need to make I9999999 trees work is also in that
+      # table
+      #
+      #########################################################################
+      :leaf_part => find_or_create_leaf_part(values[1]),
+      :pm_type => values[3]
+    })
+    attempt_to_overwrite!(@pm)
+
+    @pm.filename = values[0]
+    @pm.code = values[1]
+    @pm.area_corr = values[2]
+    begin
+      @pm.date = Date.strptime(values[4], "%d/%m/%y")
+    rescue ArgumentError
+      @pm.date = nil
```

Comment:
You say in the model that you need to validate the time format, it's probably worth validating the date format too. So I think it's best not to rescue these errors, let them propagate.

Presumably this is done to allow an empty date / time to be entered? In which case I'd do something like:

    @pm.date = Date.strptime(values[4], "%d/%m/%y")           unless values[4].nil?

The fact that they might not provide a time is concerning really. The best case scenario would be to model this is as a ruby `DateTime` and combine these into the same field. But then you would end up with a lot of entries at midnight if people hadn't entered a time...

Has the team said they want to be able to enter data without times or is it just the sample data that looks like that?